### ISSUE TITLE
[Q] Early Stopping - Unclear documentation. What is an iteration?

### ISSUE THREAD

**Issue by sdtblck** [wandb/client#2367]
When using wandb.sweep to do early stopping, what *exactly* do min_iter / max_iter specify? The documentation is very unclear, and from looking at the code and testing with various settings I still can't determine what exactly an 'iter' is.

![Screenshot from 2021-07-06 21-59-34](https://user-images.githubusercontent.com/46172032/124659864-b73b0280-dea5-11eb-8112-9e2d5736562c.png)

The docs say it's based on 'logging' iterations for the metric - so if i have a metric that I log say,  only every 100 model steps, but when calling wandb.log I accurately report the model step count, like so:

```
wandb.log(loss, step=100)
```

and then 100 steps later

```
wandb.log(loss, step=200)
```

is the number of iterations here (as hyperband sees it) - 2, or 200?

**Comment by tyomhak** [comment]
Hey @sdtblck , let me ask our sweeps engineer to double check.

**Comment by aidanjdonohue** [comment]
@stdblck I apologize for the delay. It is the number of times you log the metric you are trying to optimize. So if I log my training loss every step and only log my validation loss every 100 steps, if I'm optimizing my validation loss then every 100 steps will be seen as a single iteration.
